site/docs/coverage.md

```diff
@@ -0,0 +1,192 @@
+  `--nocache_test_results` can be set specifically for coverage runs,
+  although this of course incurs a heavy cost in terms of test times.
+
+## Language-specific configuration
+
+### Java
+
+Java should work out-of-the-box with the default configuration. The
+[bazel toolchains](https://github.com/bazelbuild/bazel-toolchains)
+contain everything necessary for remote execution, as well, including
+JUnit.
+
+[lcov]: https://github.com/linux-test-project/lcov
+
+### Python
+
+#### Prerequisites
+
+Running coverage with python requires some prerequisites:
+
+- A bazel binary that includes [b01c859](https://github.com/bazelbuild/bazel/commit/b01c85962d88661ec9f6c6704c47d8ce67ca4d2a), `bazel-6.0.0` currently has this feature already.
+- A [modified version of coverage.py](https://github.com/ulfjack/coveragepy/tree/lcov-support).
+
+#### Consuming the modified coverage.py
+
+A way to do this is via [rules_python](https://github.com/bazelbuild/rules_python), this provides the ability to use a `requirements.txt` file, the requirements listed in the file are then created as bazel targets using the [pip_install](https://github.com/bazelbuild/rules_python#installing-pip-dependencies) repository rule.
+
+The `requirements.txt` should have the following entry:
+
+```text
+git+https://github.com/ulfjack/coveragepy.git@lcov-support
+```
+
+The `rules_python`, `pip_install`, and the `requirements.txt` file should then be used in the WORKSPACE file as:
+
+```python
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
+    name = "rules_python",
+    url = "https://github.com/bazelbuild/rules_python/releases/download/0.5.0/rules_python-0.5.0.tar.gz",
+    sha256 = "cd6730ed53a002c56ce4e2f396ba3b3be262fd7cb68339f0377a45e8227fe332",
+)
+
+load("@rules_python//python:pip.bzl", "pip_install")
+
+pip_install(
+   name = "python_deps",
+   requirements = "//:requirements.txt",
+)
+```
+
+The coverage.py requirement can then be consumed with the following in the `BUILD` file:
+
+```python
+load("@python_deps//:requirements.bzl", "entry_point")
+
+alias(
+    name = "python_coverage_tools",
+    actual = entry_point("coverage"),
+)
+
+py_test(
+    name = "test",
+    srcs = ["test.py"],
+    env = {
+        "PYTHON_COVERAGE": "$(location :python_coverage_tools)",
+    },
+    deps = [
+        ":main",
+        ":python_coverage_tools",
+    ],
+)
+```
+
+#### Execution of bazel coverage
+
+To execute the coverage, use the following command:
+`bazel coverage //src:test`
+
+The resulting coverage can then be found in `$(bazel info output_path)/_coverage/_coverage_report.dat`.
+
+If you only care about local execution, you can remove the boilerplate for dynamically genrating PYTHON_COVERAGE individually via the env property of each test case and instead set it to the workspace-relative output path for coverage for all test targets by setting --test_env=PYTHON_COVERAGE=path/to/coverage.
+
+#### Extra flags for remote execution in Python
```

Comment:
Could you document what errors one may get if these flags aren't set?

Reply:
Ping.

Reply:
Yep, aware, just need to run it all again :) I haven't run the python workflow thoroughly myself yet, and our action history was deleted by the GitHub outages.

Reply:
Surprisingly, none of these flags are "extra". They're the same as needed for remote execution in general, we had some merge issues in our testing repo that made us lose track of how they interact (and well, one flag was duplicated).

I've linked the upstream issue for both, which shows all the errors you run into.